pkg_resources/tests/test_pkg_resources.py

```diff
@@ -190,6 +191,83 @@ def test_setuptools_not_imported(self):
         subprocess.check_call(cmd)
 
 
+def touch_file(path):
+    """
+    Create an empty file.
+    """
+    with open(path, 'w') as f:
+        f.write('')
+
+
+def make_distribution_no_version(tmpdir, basename):
+    """
+    Create a distribution directory with no file containing the version.
+    """
+    # Convert the LocalPath object to a string before joining.
+    dist_dir = os.path.join(str(tmpdir), basename)
+    os.mkdir(dist_dir)
+    # Make the directory non-empty so distributions_from_metadata()
+    # will detect it and yield it.
+    touch_file(os.path.join(dist_dir, 'temp.txt'))
+
+    dists = list(pkg_resources.distributions_from_metadata(dist_dir))
+    assert len(dists) == 1
+    dist, = dists
+
+    return dist, dist_dir
+
+
+def check_version_missing(dist, metadata_path, expected_text,
```

Comment:
I think rename this to `assert_version_missing`, since this is essentially a fancy type of assertion.